tables/endpointsecurity/src/fileeventstableplugin.cpp

```diff
@@ -0,0 +1,171 @@
+#include "fileeventstableplugin.h"
+
+#include <chrono>
+#include <mutex>
+
+
+namespace zeek {
+struct FileEventsTablePlugin::PrivateData final {
+  PrivateData(IZeekConfiguration &configuration_, IZeekLogger &logger_)
+      : configuration(configuration_), logger(logger_) {}
+
+  IZeekConfiguration &configuration;
+  IZeekLogger &logger;
+
+  RowList row_list;
+  std::mutex row_list_mutex;
+  std::size_t max_queued_row_count{0U};
+};
+
+Status FileEventsTablePlugin::create(Ref &obj,
+                                     IZeekConfiguration &configuration,
+                                     IZeekLogger &logger) {
+  obj.reset();
```

Comment:
Yes, you are correct about precondition. When I was writing `fileeventstableplugin.cpp` I just followed the pattern used in the existing `processeventstableplugin.cpp` file. But this `reset()` is actually redundant since function `create` will only be called once and there is no point in resetting `obj`. I will remove `reset` from both in `fileeventstableplugin.cpp` and `processeventstableplugin.cpp`.

Reply:
I have removed obj.reset() in my recent commits.